numba/core/byteflow.py

```diff
@@ -462,30 +492,87 @@ def op_LOAD_DEREF(self, state, inst):
         state.push(res)
 
     def op_LOAD_CONST(self, state, inst):
-        res = state.make_temp("const")
+        # append const index for interpreter to read the const value
+        res = state.make_temp("const") + f".{inst.arg}"
         state.push(res)
         state.append(inst, res=res)
 
     def op_LOAD_ATTR(self, state, inst):
         item = state.pop()
-        if PYVERSION in ((3, 12), ):
+        res = state.make_temp()
+        if PYVERSION in ((3, 13),):
+            state.push(res)  # the attr
             if inst.arg & 1:
                 state.push(state.make_null())
+        elif PYVERSION in ((3, 12),):
+            if inst.arg & 1:
+                state.push(state.make_null())
+            state.push(res)
         elif PYVERSION in ((3, 10), (3, 11)):
-            pass
+            state.push(res)
         else:
             raise NotImplementedError(PYVERSION)
-        res = state.make_temp()
         state.append(inst, item=item, res=res)
-        state.push(res)
 
     def op_LOAD_FAST(self, state, inst):
-        name = state.get_varname(inst)
+        assert PYVERSION <= (3, 13)
+        if PYVERSION in [(3, 13)]:
+            try:
+                name = state.get_varname(inst)
+            except IndexError:   # oparg is out of range
+                # Handle this like a LOAD_DEREF
+                # Assume MAKE_CELL and COPY_FREE_VARS has correctly setup the
+                # states.
+                # According to https://github.com/python/cpython/blob/9ac606080a0074cdf7589d9b7c9413a73e0ddf37/Objects/codeobject.c#L730C9-L759 # noqa E501
+                # localsplus is locals + cells + freevars
+                bc = state._bytecode
+                num_varnames = len(bc.co_varnames)
+                num_freevars = len(bc.co_freevars)
+                num_cellvars = len(bc.co_cellvars)
+                max_fast_local = num_cellvars + num_freevars
+                assert 0 <= inst.arg - num_varnames < max_fast_local
+                res = state.make_temp()
+                state.append(inst, res=res, as_load_deref=True)
+                state.push(res)
+                return
+        else:
+            name = state.get_varname(inst)
         res = state.make_temp(name)
         state.append(inst, res=res)
         state.push(res)
 
-    if PYVERSION in ((3, 12), ):
+    if PYVERSION in ((3, 13),):
```

Comment:
Are these `LOAD` and `STORE` contractions missing from the documentation at:https://docs.python.org/3/library/dis.html ? I found a few references and they seem to be called "Super-instructions":

https://github.com/faster-cpython/ideas/issues/16

Does it make sense to raise the missing documentation on the cpython tracker?

Reply:
Yes, they are, and right there should be docs